### PR TITLE
add get/list probe command

### LINKED ISSUE #176: Will the litmusctl to support create/list/delete probe?
I don't find the command line support to create the probes. I have to create them via UI. It prevents to me create a bash script to automate the testing. Please suggest if there is a plan. I am happy to submit a PR if I know where to check the graphql api specs for probes.

### PR DESCRIPTION
**Pull request by shivam-Purohit**
closes #176

This will add a new command to litmusctl that will enable the user to get the created probes in a given project

### CODE REVIEW

**Comment by aryan-bhokare** [comment]
> there seems to be some permission issue in pre-commit file @aryan-bhokare can you check that out?

Yep I'll check that. We have to manually give permissions I'll check to automate it.

**Comment by shivam-Purohit** [comment]
two things:
1. I cannot get the last row to align properly. I dont know exactly how to do that, i tried a bunch of combination
2. the probes arent getting filtered based on probeType I treid k8sprobe filter but its giving me all the http type ones. idk why

**Comment by shivam-Purohit** [comment]
```
./litmusctl get probes --project-id="8adf62d5-64f8-4c66-ab53-63729db9dd9a"

✔ Yes

You chose "Yes"

✔ cmdProbe

✔ done

Selected Probe Types: [cmdProbe]

PROBE ID		 PROBE TYPE		 CREATED AT		 CREATED BY

http probe test		httpProbe		admin		56047-11-12 19:12:28 +0530 IST

shivam			httpProbe		admin		56053-01-12 22:58:59 +0530 IST

shivamtest		httpProbe		admin		56140-09-13 09:17:28 +0530 IST

newhttp		httpProbe		admin		56142-12-05 08:29:41 +0530 IST

✔ Press Enter to show more probes (or type 'q' to quit): q█


```

```
./litmusctl get probes --project-id="8adf62d5-64f8-4c66-ab53-63729db9dd9a" --non-interactive=true --probe-types="k8sProbe, promProbe"

PROBE ID		 PROBE TYPE		 CREATED AT		 CREATED BY

http probe test		httpProbe		admin		56047-11-12 19:12:28 +0530 IST

shivam			httpProbe		admin		56053-01-12 22:58:59 +0530 IST

shivamtest		httpProbe		admin		56140-09-13 09:17:28 +0530 IST

newhttp		httpProbe		admin		56142-12-05 08:29:41 +0530 IST

Press Enter to show more probes (or type 'q' to quit): q
```

**Comment by shivam-Purohit** [comment]
```
./litmusctl get probes --project-id="8adf62d5-64f8-4c66-ab53-63729db9dd9a" 

✔ Yes

You chose "Yes"

✔ cmdProbe

✔ promProbe

✔ done

Selected Probe Types: [cmdProbe promProbe]

PROBE ID		 PROBE TYPE		 CREATED AT		 CREATED BY

No more probes to display
```
```
./litmusctl get probes --project-id="8adf62d5-64f8-4c66-ab53-63729db9dd9a" 

✔ Yes

You chose "Yes"

✔ httpProbe

✔ promProbe

✔ done

Selected Probe Types: [httpProbe promProbe]

PROBE ID		 PROBE TYPE		 CREATED AT		 CREATED BY

http probe test		httpProbe		admin		56047-11-12 19:12:28 +0530 IST

shivamtest		httpProbe		admin		56140-09-13 09:17:28 +0530 IST

newhttp		httpProbe		admin		56142-12-05 08:29:41 +0530 IST

Press Enter to show more probes (or type 'q' to quit): 

No more probes to display
```

File: pkg/cmd/get/get.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ var GetCmd = &cobra.Command{
 		#get list of Chaos Experiment runs
 		litmusctl get chaos-experiment-runs --project-id=""
 
+		#get list of Probes in a Project
+		litmusctl list probes --project-id=""
```
Comment (Nageshbansal) [inline pkg/cmd/get/get.go:39]: it should be `litmusctl get probes`

File: pkg/cmd/get/probe.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	models "github.com/litmuschaos/litmus/chaoscenter/graphql/server/graph/model"
+	apis "github.com/litmuschaos/litmusctl/pkg/apis/probe"
+	"github.com/litmuschaos/litmusctl/pkg/utils"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+)
+
+var probesCmd = &cobra.Command{
+	Use:   "probes",
+	Short: "Display list of probes",
```
Comment (Nageshbansal) [inline pkg/cmd/get/probe.go:34]: maybe we can write this: "Display a list or a specific probe"
WDYT @SarthakJain26 ?

**Review by Nageshbansal — CHANGES_REQUESTED** [review]

File: pkg/cmd/get/probe.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+			utils.White_B.Fprintln(writer, "PROBE DETAILS")
+			utils.White.Fprintln(writer, "PROBE ID\t", probeGetData.Name)
+			utils.White.Fprintln(writer, "PROBE DESCRIPTION\t", *probeGetData.Description)
+			utils.White.Fprintln(writer, "PROBE TYPE \t", probeGetData.Type)
+			utils.White.Fprintln(writer, "PROBE INFRASTRUCTURE TYPE \t", probeGetData.InfrastructureType)
+			utils.White.Fprintln(writer, "CREATED AT\t", createdTime)
+			utils.White.Fprintln(writer, "CREATED BY\t", probeGetData.CreatedBy.Username)
+			utils.White.Fprintln(writer, "UPDATED AT\t", updatedTime)
+			utils.White.Fprintln(writer, "UPDATED BY\t", probeGetData.UpdatedBy.Username)
+			utils.White.Fprintln(writer, "TAGS\t", strings.Join(probeGetData.Tags, ", "))
```
[the remark this answers is not in the document]
Comment (shivam-Purohit) [reply pkg/cmd/get/probe.go:332]: printed all the properties for different probe types

File: pkg/cmd/get/probe.go (hunk 2 above)
Comment (SarthakJain26) [reply pkg/cmd/get/probe.go:34]: How about "Details of a specific probe" we can avoid using "list"

**Review by SarthakJain26 — CHANGES_REQUESTED** [review]

File: pkg/cmd/get/probe.go (hunk 2 above)
Comment (Nageshbansal) [reply pkg/cmd/get/probe.go:34]: yep, sounds good

File: pkg/cmd/get/probe.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nageshbansal) [reply pkg/cmd/get/probe.go:332]: could you please share a screenshot for this

File: pkg/cmd/get/probe.go (hunk 2 above)
Comment (shivam-Purohit) [reply pkg/cmd/get/probe.go:35]: but we are now majorly listing the probes and giving few details if passed probe_id. We are using describe for that right?

File: pkg/cmd/get/probe.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (shivam-Purohit) [reply pkg/cmd/get/probe.go]:
```
./litmusctl get probes --project-id="8adf62d5-64f8-4c66-ab53-63729db9dd9a" --probe-id="shivamcmd"

PROBE DETAILS

PROBE ID 			 shivamcmd

PROBE DESCRIPTION 		 

PROBE TYPE 			 cmdProbe

PROBE INFRASTRUCTURE TYPE 	 Kubernetes

TIMEOUT 			 1s

INTERVAL 			 1s

CREATED AT			 56175-10-19 08:28:39 +0530 IST

CREATED BY			 admin

UPDATED AT			 56175-10-19 08:28:39 +0530 IST

UPDATED BY			 admin

TAGS				 

./litmusctl get probes --project-id="8adf62d5-64f8-4c66-ab53-63729db9dd9a"

✔ No

You chose "No"

PROBE ID		 PROBE TYPE		 CREATED AT		 CREATED BY

http probe test		httpProbe		admin		56047-11-12 19:12:28 +0530 IST

newhttp		httpProbe		admin		56142-12-05 08:29:41 +0530 IST

shivam99		httpProbe		admin		56175-07-01 21:35:07 +0530 IST

shivamcmd		cmdProbe		admin		56175-10-19 08:28:39 +0530 IST

shivamcmd1		cmdProbe		admin		56176-02-14 18:40:30 +0530 IST

Press Enter to show more probes (or type 'q' to quit): q
```

File: pkg/cmd/get/probe.go [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+			fmt.Scanln(&projectID)
+
+			if projectID == "" {
+				utils.Red.Println("⛔ Project ID can't be empty!!")
+				os.Exit(1)
+			}
+		}
+		ProbeID, err := cmd.Flags().GetString("probe-id")
+
+		if ProbeID == "" {
```
[the remark this answers is not in the document]
Comment (shivam-Purohit) [reply pkg/cmd/get/probe.go:56]: done PTAL

**Comment by shivam-Purohit** [comment]
its failing as soon as i rebased the master. But was working fine on local. what should i do?

File: pkg/apis/probe/query.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	GetProbeQuery = `query getProbe($projectID: ID!, $probeName: ID!) {
+		getProbe(projectID: $projectID, probeName: $probeName) {
+		  name
+		  description
+		  type
+		  infrastructureType
+		  kubernetesHTTPProperties{
+			probeTimeout
+			interval
+			retry
```
Comment (Nageshbansal) [inline pkg/apis/probe/query.go:24]: Are we using all the properties? If not, then we can consider removing them from query

File: pkg/cmd/get/probe.go [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+	probes_get, _ := apis.ListProbeRequest(projectID, selectedItems, credentials)
+	probes_data := probes_get.Data.Probes
+
+	itemsPerPage := 5
+	page := 1
+	totalProbes := len(probes_data)
+
+	writer := tabwriter.NewWriter(os.Stdout, 8, 8, 8, '\t', tabwriter.AlignRight)
+	utils.White_B.Fprintln(writer, "PROBE ID\t PROBE TYPE\t CREATED BY\t CREATED AT")
```
Comment (Nageshbansal) [inline pkg/cmd/get/probe.go:141]: Let's have `referencedBy` also here ( as we also show this UI Overview )

Comment (shivam-Purohit) [reply pkg/cmd/get/probe.go]: done you can take a look

**Review by SarthakJain26 — APPROVED** [review]

**Review by Nageshbansal — APPROVED** [review]

**Pull request merged**